File: frontend/bhau_fitness_flutter/lib/models/plan.dart
class Plan {
  final int id;
  final String name;
  final double price;
  final int durationDays;
  final String? description;

  Plan({
    required this.id,
    required this.name,
    required this.price,
    required this.durationDays,
    this.description,
  });

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      id: json['id'] as int,
      name: json['name'] as String,
      price: (json['price'] as num).toDouble(),
      durationDays: json['durationDays'] as int,
      description: json['description'] as String?,
    );
  }
}

File: backend/BhauFitnessApi/Models/Entities/Plan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BhauFitnessApi.Models.Entities;

public class Plan : IMultitenant
{
    public int Id { get; set; }
    public string TenantId { get; set; } = "default";

    [Required, MaxLength(50)]
    public string Name { get; set; } = string.Empty; // "Basic", "Premium", "Elite"

    [Column(TypeName = "decimal(10,2)")]
    public decimal Price { get; set; } // monthly price in INR

    public int DurationDays { get; set; } = 30;

    public bool IsActive { get; set; } = true;

    [MaxLength(500)]
    public string? Description { get; set; }

    public ICollection<Membership> Memberships { get; set; } = new List<Membership>();
}
